src/query_error.c

```diff
@@ -52,6 +52,7 @@ void QueryError_SetError(QueryError *status, QueryErrorCode code, const char *er
   } else {
     status->detail = rm_strdup(QueryError_Strerror(code));
   }
+  status->message = status->detail;
```

Comment:
Here err is const char*, which indicates it wasn't allocated on the heap.
So it is a constant string and so no user data can be inside of it.
`message` is also a const char* and does not get freed.
Only `detail` gets freed so no double free can happen.

If someone formats a string and passes it to this function then we will have lifetime management issues since we might use the string while the entity holding the memory will release it.
So from a memory management pov it wouldn't make a lot of sense.

Is it possible? sure but not very likely.